lib/engines/nunjucks.js

```diff
@@ -399,4 +400,45 @@ export class NunjucksEngine extends BaseEngine {
       });
     });
   }
+
+  async watch(fn) {
+    const toWatch = Array.from(this.dependencies);
+
+    this.watcher = chokidar.watch(toWatch, {
+      ignoreInitial: true,
+    });
+
+    const onChange = async (path) => {
+
+      // find the files to work with
+      const files = await this.findFiles();
+
+      try {
+        for (const file of files) {
+          if (file !== path)
+            continue;
+          logger('┌╼ rendering', relative(this.input, file));
+          await this.outputFile(file);
+        }
+        // if we have a dynamic generator, let's find them
+        if (this.createPages) {
+          const { collect, renders } = this.collectDynamicRenders();
+          await this.createPages(collect, this.context);
+
+          for (const [inputPath, outputPath, localContext] of renders) {
+            if (inputPath !== path)
+              continue;
+            logger('┌╼ dynamically rendering', outputPath);
+            await this.outputFile(inputPath, outputPath, localContext);
+          }
+        }
+      } catch (err) {
+        throw err;
+      }
```

Comment:
Can you explain to me the need for this catch?

Reply:
I see now it's the same behavior as [the "super" function](https://github.com/datadesk/baker/blob/main/lib/engines/base.js#L156-L174). Is that the idea?

Reply:
Yes exactly. It's the behavior from up above, but I wasn't sure exactly how this stuff was *supposed* to work so I tried to make it clear if it failed for some reason. I supposed we could also do a `noop` or something, but I wasn't sure if that would have unintended side-effects.

Reply:
Thank you.